File: unix/tooltip.c
#include "uipriv_unix.h"

void uiControlSetTooltip(uiControl *c, const char *tooltip)
{
	GtkWidget *widget;

	widget = GTK_WIDGET(uiControlHandle(c));
	if (c->TypeSignature == uiSliderSignature) {
		uiSlider *s = uiSlider(c);

		if (tooltip != NULL)
			uiprivSliderSetControlTooltip(s, 1);
		gtk_widget_set_tooltip_text(widget, tooltip);
		if (tooltip == NULL)
			uiprivSliderSetControlTooltip(s, 0);
		return;
	}

	gtk_widget_set_tooltip_text(widget, tooltip);
}

File: darwin/tooltip.m
#include "uipriv_darwin.h"

// https://developer.apple.com/documentation/appkit/nsview/1483541-tooltip?language=objc

void uiControlSetTooltip(uiControl *c, const char *tooltip)
{
	NSView *view;

	if (c->TypeSignature == uiWindowSignature)
		view = [(NSWindow *)uiControlHandle(c) contentView];
	else
		view = (NSView *)uiControlHandle(c);

	if (c->TypeSignature == uiSliderSignature && tooltip != NULL)
		uiprivSliderSetControlTooltip(uiSlider(c), 1);
	[view setToolTip:(tooltip == NULL ? nil : uiprivToNSString(tooltip))];
	if (c->TypeSignature == uiSliderSignature && tooltip == NULL)
		uiprivSliderSetControlTooltip(uiSlider(c), 0);
}
